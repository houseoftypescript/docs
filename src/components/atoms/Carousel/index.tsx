import ChevronLeft from '@mui/icons-material/ChevronLeft';
import ChevronRight from '@mui/icons-material/ChevronRight';
import React, { useState } from 'react';
import { CDN_IMAGE_URL } from '../../../constants';

export const Carousel: React.FC = () => {
  const [count, setCount] = useState(0);

  return (
    <div
      className="relative w-full overflow-hidden"
      style={{ paddingBottom: '50%' }}
    >
      {[3, 2, 1].map((intro, index) => {
        const backgroundImage = `url(${CDN_IMAGE_URL}/intros/intro-${intro}.png)`;
        const zIndex = Math.abs((index + count) % 3) * 10;
        return (
          <div
            key={`intro-${intro}`}
            className={`absolute w-full h-full bg-cover bg-center transition-all top-0 left-0`}
            style={{ zIndex, backgroundImage }}
          >
            <div className="w-full h-full bg-gray-100/20" />
          </div>
        );
      })}
      <div
        className="z-40 absolute left-0 top-0 bottom-0 m-auto w-8 h-16 bg-gray-900/50 hover:bg-gray-900/75 cursor-pointer text-white flex items-center justify-center"
        onClick={() => {
          setCount((prev) => prev - 1);
        }}
      >
        <ChevronLeft className="text-3xl" />
      </div>
      <div
        className="z-40 absolute right-0 top-0 bottom-0 m-auto w-8 h-16 bg-gray-900/50 hover:bg-gray-900/75 cursor-pointer text-white flex items-center justify-center"
        onClick={() => {
          setCount((prev) => prev + 1);
        }}
      >
        <ChevronRight className="text-3xl" />
      </div>
    </div>
  );
};

Carousel.displayName = 'Carousel';
Carousel.defaultProps = {};

export default Carousel;
