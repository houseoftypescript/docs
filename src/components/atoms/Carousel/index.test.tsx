import Carousel from '.';
import { render } from '@testing-library/react';

describe('Carousel', () => {
  test('render default', () => {
    const wrapper = render(<Carousel />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
