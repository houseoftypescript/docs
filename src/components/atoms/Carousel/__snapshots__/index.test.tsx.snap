// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel render default 1`] = `
<div>
  <div
    class="relative w-full overflow-hidden"
    style="padding-bottom: 50%;"
  >
    <div
      class="absolute w-full h-full bg-cover bg-center transition-all top-0 left-0"
      style="z-index: 0; background-image: url(https://raw.githubusercontent.com/hieudoanm/cdn/master/images/vleague/intros/intro-3.png);"
    >
      <div
        class="w-full h-full bg-gray-100/20"
      />
    </div>
    <div
      class="absolute w-full h-full bg-cover bg-center transition-all top-0 left-0"
      style="z-index: 10; background-image: url(https://raw.githubusercontent.com/hieudoanm/cdn/master/images/vleague/intros/intro-2.png);"
    >
      <div
        class="w-full h-full bg-gray-100/20"
      />
    </div>
    <div
      class="absolute w-full h-full bg-cover bg-center transition-all top-0 left-0"
      style="z-index: 20; background-image: url(https://raw.githubusercontent.com/hieudoanm/cdn/master/images/vleague/intros/intro-1.png);"
    >
      <div
        class="w-full h-full bg-gray-100/20"
      />
    </div>
    <div
      class="z-40 absolute left-0 top-0 bottom-0 m-auto w-8 h-16 bg-gray-900/50 hover:bg-gray-900/75 cursor-pointer text-white flex items-center justify-center"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-3xl css-i4bv87-MuiSvgIcon-root"
        data-testid="ChevronLeftIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
    </div>
    <div
      class="z-40 absolute right-0 top-0 bottom-0 m-auto w-8 h-16 bg-gray-900/50 hover:bg-gray-900/75 cursor-pointer text-white flex items-center justify-center"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-3xl css-i4bv87-MuiSvgIcon-root"
        data-testid="ChevronRightIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </div>
  </div>
</div>
`;
